lib/route.js

```diff
@@ -208,25 +214,34 @@ function buildRouting (options) {
         this[kReplySerializerDefault]
       )
 
-      // TODO this needs to be refactored so that buildSchemaCompiler is
-      // not called for every single route. Creating a new one for every route
-      // is going to be very expensive.
       if (opts.schema) {
-        if (this[kSchemaCompiler] == null && this[kSchemaResolver]) {
-          done(new FST_ERR_SCH_MISSING_COMPILER(opts.method, url))
+        try {
+          context.schema = normalizeSchema(context.schema)
+        } catch (error) {
+          done(error)
           return
         }
 
+        const sBucket = this[kSchemas]
+        if (!opts.validatorCompiler && !this[kValidatorCompiler]) {
```

Comment:
@delvedor I keep the same behaviour discussed here https://github.com/fastify/fastify/issues/2006 

What I had in mind is that we could call `this.setValidatorCompiler(buildDefaultValidator(sBucket.getSchemas(), this[kOptions].ajv, sBucket.cache))` with the server startup, instead in the first route inizialization, but this would mean that users that don't use schemas, will have unsuefull ajv instances in memory.